src/include/mallocMC/mallocMC_utils.hpp

```diff
@@ -186,4 +186,39 @@ namespace mallocMC
   template<class T>
   MAMC_HOST MAMC_ACCELERATOR inline T divup(T a, T b) { return (a + b - 1)/b; }
 
+  /** the maximal number threads per block
+   *
+   * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities
+   */
+  struct MaxThreadsPerBlock
+  {
+	// valid for sm_2.X - sm_7.5
+    BOOST_STATIC_ASSERT uint32_t value = 1024;
+  };
+
+  /** the maximal number threads per block
+   *
+   * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities
+   */
+  struct WarpSize
+  {
+	// valid for sm_2.X - sm_7.5
+    BOOST_STATIC_ASSERT uint32_t value = 32;
+  };
+
+  /** warp id within a block
+   *
+   * The id is constant over the livetime of the thread.
+   * The id is not equal to warpid().
+   *
+   * @return warp id within the block
```

Comment:
in `warpid()` we should add a comment that it should not be used besides for diagnostics.
Maybe we want to implement the [`%ctaid and %tid` getter](https://github.com/ComputationalRadiationPhysics/mallocMC/pull/149#issuecomment-451088827) as described by the docs.

Can you please `grep` the code after your change to check no `warpid` is used anywhere else?

Reply:
> in warpid() we should add a comment that it should not be used besides for diagnostics.

We can extend the documentation but the point with the diagnostic is not correct e.g. `warpid` is still used for a hashing function where it makes sense and it is not critical. 

> Maybe we want to implement the %ctaid and %tid getter as described by the docs.

I do not understand how this helps because `%%tid == threadIdx`